openedx/core/djangoapps/user_api/api/profile.py

```diff
@@ -151,22 +101,19 @@ def update_email_opt_in(username, org, optin):
         None
 
     Raises:
-        ProfileUserNotFound: Raised when the username specified is not associated with a user.
+        AccountUserNotFound: Raised when the username specified is not associated with a user.
 
     """
-    try:
-        user = User.objects.get(username=username)
-    except User.DoesNotExist:
-        raise ProfileUserNotFound
-
-    profile = UserProfile.objects.get(user=user)
+    account_settings = AccountView.get_serialized_account(username)
+    year_of_birth = account_settings['year_of_birth']
     of_age = (
-        profile.year_of_birth is None or  # If year of birth is not set, we assume user is of age.
```

Comment:
I think we will want to change this assumption (year of birth not set, assume user of age). We certainly plan to take the opposite approach for profile pictures. But that is beyond the scope of this PR....